### PR TITLE
Db migration: remove gene length

### PR DESCRIPTION
**Pull request by inodb**
This information is stored in Genome Nexus. We don't need it in
cBioPortal anymore.

Fix https://github.com/cBioPortal/cbioportal/issues/6308

- [x] Verify that it is no longer used in frontend? (it was used in Mutated Genes tab of study view)
- [x] Remove LENGTH in gene table
- [x] Remove it from API
- [x] Remove it from import code

### CODE REVIEW

**Comment by inodb** [comment]
@sheridancbio i forgot what the rules are regarding schema versioning. Where can I find that again?

File: core/src/main/java/org/mskcc/cbio/portal/servlet/PfamSequenceServlet.java [hunk 1]
```diff
@@ -1,259 +0,0 @@
-/*
```
Comment (inodb) [inline core/src/main/java/org/mskcc/cbio/portal/servlet/PfamSequenceServlet.java]: don't think this is used anymore either, all pfam data is in genome nexus

File: portal/src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -362,15 +362,6 @@
         <url-pattern>/omaRedirect.do</url-pattern>
     </servlet-mapping>
 
-	<servlet>
```
Comment (inodb) [inline portal/src/main/webapp/WEB-INF/web.xml]: don't think this is used anymore either, all pfam data is in genome nexus

**Comment by inodb** [comment]
Thanks @pieterlukasse ! Good catch! Updated

That reminds me we might have to do a mouse version of genome nexus:

https://github.com/genome-nexus/genome-nexus-importer/issues/22

I don't think mouse support for cBioPortal will work currently

File: docs/Updating-gene-and-gene_alias-tables.md [hunk 3]
```diff
@@ -39,27 +39,23 @@ ALTER TABLE `geneset` AUTO_INCREMENT = 1;
...
 
-5- You probably also want to update the gene lengths. To do so, download .annotation.gtf.gz of the [latest GENCODE release for hg19 / GRCh37](http://www.gencodegenes.org/releases/26lift37.html) for human and [mm10 / GRCm38](http://www.gencodegenes.org/mouse_releases/current.html) for mouse.
-
-After downloading, go to your downloads directory, decompress the file and add it as an argument (--gtf) in the next step.
-
-6- To import gene data type the following commands when in the folder `<cbioportal_source_folder>/core/src/main/scripts`:
+5- To import gene data type the following commands when in the folder `<cbioportal_source_folder>/core/src/main/scripts`:
 ```
 export PORTAL_HOME=<cbioportal_configuration_folder>
 ./importGenes.pl --genes <ncbi_species.gene_info> --gtf <gencode.v25.annotation.gtf>
```
Comment (ao508) [inline docs/Updating-gene-and-gene_alias-tables.md]: update args (remove `--gtf`)

File: web/src/test/java/org/cbioportal/web/DiscreteCopyNumberControllerTest.java [hunk 4]
```diff
@@ -139,7 +139,6 @@ public void getDiscreteCopyNumbersInMolecularProfileBySampleListIdDetailedProjec
             .andExpect(MockMvcResultMatchers.jsonPath("$[0].gene.hugoGeneSymbol").value(TEST_HUGO_GENE_SYMBOL_1))
             .andExpect(MockMvcResultMatchers.jsonPath("$[0].gene.type").value(TEST_TYPE_1))
             .andExpect(MockMvcResultMatchers.jsonPath("$[0].gene.cytoband").value(TEST_CYTOBAND_1))
-            .andExpect(MockMvcResultMatchers.jsonPath("$[0].gene.length").value(TEST_LENGTH_1))
```
Comment (ao508) [inline web/src/test/java/org/cbioportal/web/DiscreteCopyNumberControllerTest.java]: `TEST_LENGTH_1` and `TEST_LENGTH_2` should be removed above (starting at L47)

File: web/src/test/java/org/cbioportal/web/GeneControllerTest.java [hunk 5]
```diff
@@ -132,7 +130,6 @@ public void getGene() throws Exception {
         gene.setHugoGeneSymbol(HUGO_GENE_SYMBOL_1);
         gene.setType(TYPE_1);
         gene.setCytoband(CYTOBAND_1);
-        gene.setLength(LENGTH_1);
```
Comment (ao508) [inline web/src/test/java/org/cbioportal/web/GeneControllerTest.java]: same comment here about removing the `LENGTH*` vars

File: web/src/test/java/org/cbioportal/web/MutationControllerTest.java [hunk 6]
```diff
@@ -283,7 +283,6 @@ public void getMutationsInMolecularProfileBySampleListIdDetailedProjection() thr
             .andExpect(MockMvcResultMatchers.jsonPath("$[0].gene.hugoGeneSymbol").value(TEST_HUGO_GENE_SYMBOL_1))
             .andExpect(MockMvcResultMatchers.jsonPath("$[0].gene.type").value(TEST_TYPE_1))
             .andExpect(MockMvcResultMatchers.jsonPath("$[0].gene.cytoband").value(TEST_CYTOBAND_1))
```
Comment (ao508) [inline web/src/test/java/org/cbioportal/web/MutationControllerTest.java]: same comment here about removing the `TEST_LENGTH*` vars

**Comment by inodb** [comment]
@ao508 	Thanks for catching those ! Updated the PR

**Comment by pvannierop** [comment]
@inodb Gene length appears to be used in the lollipop plot component.
```
  private get proteinLength(): number {
        return (this.props.store.allTranscripts.result &&
            this.props.store.activeTranscript &&
            this.props.store.transcriptsByTranscriptId[this.props.store.activeTranscript] &&
            this.props.store.transcriptsByTranscriptId[this.props.store.activeTranscript].proteinLength) ||
            Math.round(this.props.store.gene.length / 3);
    }
```

**Comment by pvannierop** [comment]
I misunderstood the idea of the PR. It was remapping of gene.length from cBP-db to GN. Never mind my comment.

**Comment by sheridancbio** [comment]
Hi @inodb  ... here were old rules we adopted for db schema version number assignment. I'm not sure what our current practice is.  https://github.com/cBioPortal/cbioportal/wiki/EP-Review-Changes-to-Database-Schema

**Pull request merged**